.github/workflows/test.yml

```diff
@@ -0,0 +1,34 @@
+name: Build
+on:
+  pull_request:
+    branches: [master]
+  push:
+    branches:
+      - master
+jobs:
+  test:
+    name: Test Suite
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        rust: [stable]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      - name: Install stable toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
```

Comment:
YMMV but this failed locally (`act --job test` actually worked!) for me with:

```h
error: 'cargo-clippy' is not installed for the toolchain 'stable-x86_64-unknown-linux-gnu'.
```

Reply:
i still cant get act to work....

Reply:
test is the testing pipeline which i have not touched much, i am touching the build pipeline here :)